### DISCUSSION TITLE
v8 - A guide in the documentation about using nuxt-i18n with nuxt-content 2

### DISCUSSION THREAD

**Discussion by Pamavoc** [nuxt-modules/i18n#1872] [Ideas]
I can't get it to work correctly with the query builder and I believe it could be great to see the "correct" implementation of nuxt-content with nuxt-i18n.

**Comment by ineshbose** [comment]
There is a bit of planning needed to see how to best integrate the content module with `@nuxtjs/i18n` based on your routing strategy, pages/components, `content/` organisation, etc.

However, the basic idea can be similar to this to get you started - but you need to tailor it according to your app:

```html
<script setup>
const { path } = useRoute()
const { locale } = useI18n()

const { data: documentContent } = await useAsyncData('doc', () => {
  // here you query your documents
  return queryContent({ _path: `${locale}/customDir/${path}` }).findOne()
})
</script>

<template>
  <ContentRenderer v-if="documentContent" :value="documentContent" />
</template>
```
